### DISCUSSION TITLE
Blocks within repeaters (or multiple block editors)

### DISCUSSION THREAD

**Discussion by fusionconsulting** [area17/twill#1770] [General]
I'm not sure whether the issue is that I'm using multiple block editors or that I'm using a block editor inside a repeater but I have a problem where the contents of both block editors is the same. Here is my form:

    <a17-fieldset title="Product data" id="data" :open="true">
        @formField('block_editor', [
            'blocks' => ['product-data'],
            'name' => 'product-data'
        ])
    </a17-fieldset>
    <a17-fieldset title="Product options" id="options" :open="true">
        @formField('repeater', ['type' => 'product-option'])
    </a17-fieldset>

And my repeater looks like this:

    @twillRepeaterTitle('Product Option')
    @twillRepeaterTrigger('Add option')
    @twillRepeaterGroup('app')
 
    @formField('input', [
        'name' => 'title',
        'label' => 'Title',
        'required' => true,
    ])
    @formField('block_editor', [
        'blocks' => ['product-option-item'],
        'name' => 'product-options'
    ])

Whenever I add a product option and add a block to it, it gets duplicated in the all the other repeater block editors. I've searched for the issue and the only thing I've found that seemed relevant was this - https://github.com/area17/twill/pull/112 which is why I added the "name" to the block editors but it made no difference.

Is there a way to accomplish this? Am I doing something wrong?

**Comment by ifox** [comment]
Hi @fusionconsulting, at the moment it is not possible to nest a block editor field within a repeater field.

**Reply by fusionconsulting** [reply]
Thanks for the reply. I managed to solve the problem by switching them around so now I have the repeater within a block and it works perfectly.
